ONSdigital/ssdc-rm-acceptance-tests
# Run exception manager tests first and increase wait time for support …

…tool tests

# Motivation and Context

The acceptance tests in concourse seem to fail occasionally for reasons unknown. 
# What has changed



There seem to be two main themes I noticed, one where pubsub messages seem to be left over on queues (and these fail early on in the tests) which to me suggests that maybe pubsub was in a cold state so the test timed out before it could retrieve the pubsub message. To try and solve this issue I've modified some of the feature file names so that behave runs the exception manager tests first. I've chosen this one because it uses pubsub and it's looking for failures anyway, and the test itself does do a lot of waiting so this should hopefully be enough time for pubsub to warm up before the test fails.

The other common theme I noticed was a lot of failures with the support tool where it can't find things such as packcodes in the UI tests. To try to reduce the failures with this I've upped the timeout where its looking for the element so this should hopefully solve any issue if the page is being slow.
# How to test?



Check they pass locally
Build it
Check they run in a GCP project (and that the exception manager tests run first)
# Links



https://trello.com/c/NLQCxrxk
# Screenshots (if appropriate):
